enumerables.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    selfitem.my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  def my_any?
+    selfitem = self
+    selfitem.my_each { |x| return true if yield(x) }
+    false
+  end
+
+  def my_none?
+    selfitem = self
+    selfitem.my_each { |x| return false if yield(x) }
+    true
+  end
+
+  def my_count
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array.length
+  end
```

Comment:
enumerables my_count returns the number of items in enum through enumeration
      Failure/Error: expect(array.my_count).to eq array.count
      
      LocalJumpError:
        no block given (yield)

enumerables my_count counts the number of items in enum that are equal to item if an argument is given
      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)